src/hazelweb/Main.re

```diff
@@ -1,10 +1,42 @@
+module Exn = Base.Exn;
+module Dom = Js_of_ocaml.Dom;
+module Vdom = Virtual_dom.Vdom;
+
 Logger.init_log();
 
-// Start the main app.
-// See <https://github.com/janestreet/incr_dom/blob/master/src/start_app.mli>.
-Incr_dom.Start_app.start(
-  (module Hazel),
-  ~debug=false,
-  ~bind_to_element_with_id="container",
-  ~initial_model=Model.init(),
-);
+let () = {
+  // Handle exceptions via a Monitor
+  // See <https://github.com/janestreet/incr_dom/blob/master/example/monitor>
+  let monitor = Async_kernel.Monitor.create();
+  let _ =
+    Async_kernel.Monitor.detach_and_iter_errors(
+      monitor,
+      ~f=exn => {
+        let message = Exn.to_string(Async_kernel.Monitor.extract_exn(exn));
+        let root = JSUtil.force_get_elem_by_id("root");
+        let error_box =
+          Vdom.Node.to_dom(
+            Vdom.Node.body(
+              [Vdom.Attr.id("error-message")],
+              [
+                Vdom.Node.h3([], [Vdom.Node.text("ERROR!")]),
+                Vdom.Node.text(message),
+              ],
+            ),
+          );
+        Dom.appendChild(root, error_box);
+      },
```

Comment:
IIUC, Hazel completely stops working if this case is triggered.

In that case, I would suggest putting the message at the top of the screen so there is no way to miss it.

I would also suggest stronger wording than "Error" because that sounds like the user made a mistake that he or she can correct.

Maybe something like "Hazel has crashed" with a message telling users to reload the page (and how to reload).  If there is a way to save and reload the user's code after refreshing, tell them how to do that.  If not, warn them that their code will be lost, and they may want to manually copy it before reloading.

Reply:
I agree that we should say "Hazel has crashed" instead fo "Error"